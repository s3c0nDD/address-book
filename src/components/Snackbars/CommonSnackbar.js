import React from 'react';
import PropTypes from 'prop-types';

import Alert from '@material-ui/lab/Alert';
import Snackbar from '@material-ui/core/Snackbar';

const CommonSnackbar = ({ action, message, open, severity, ...props }) => {
  return (
    <Snackbar
      {...props}
      open={open}
    >
      <Alert
        variant="filled"
        severity={severity}
        action={action}
      >
        {message}
      </Alert>
    </Snackbar>
  )
};

export default CommonSnackbar;

CommonSnackbar.propTypes = {
  action: PropTypes.element,
  message: PropTypes.string.isRequired,
  open: PropTypes.bool,
  severity: PropTypes.oneOf(['success', 'error', 'info', 'warning']),
  props: PropTypes.arrayOf(PropTypes.any)
};