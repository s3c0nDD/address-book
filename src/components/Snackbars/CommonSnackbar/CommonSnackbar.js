import React from 'react';
import PropTypes from 'prop-types';

import Alert from '@material-ui/lab/Alert';
import Snackbar from '@material-ui/core/Snackbar';

/**
 * Common snackbar component
 * @param {object} props component's props
 * @param {function} props.action react component to show in action area
 * @param {string} props.message message to show
 * @param {boolean} props.open snackbar open state
 * @param {string} props.severity snackbar severity
 */
const CommonSnackbar = ({ action, message, open, severity, ...props }) => {
  return (
    <Snackbar
      {...props}
      open={open}
    >
      <Alert
        variant="filled"
        severity={severity}
        action={action}
      >
        {message}
      </Alert>
    </Snackbar>
  )
};

export default CommonSnackbar;

CommonSnackbar.propTypes = {
  action: PropTypes.element,
  message: PropTypes.string.isRequired,
  open: PropTypes.bool,
  severity: PropTypes.oneOf(['success', 'error', 'info', 'warning']),
  props: PropTypes.arrayOf(PropTypes.any)
};