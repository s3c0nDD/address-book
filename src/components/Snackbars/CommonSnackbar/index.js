import CommonSnackbar from './CommonSnackbar'

export default CommonSnackbar;