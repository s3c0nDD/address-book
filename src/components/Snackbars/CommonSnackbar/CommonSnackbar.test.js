import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import Alert from '@material-ui/lab/Alert';
import Snackbar from '@material-ui/core/Snackbar';

import CommonSnackbar from './CommonSnackbar';

describe('CommonSnackbar', () => {
  let shallow;
  let component;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    component = (
      <CommonSnackbar
        message="Hello world"
        open={true}
        severity="info"
      />
    );
    wrapper = shallow(component);
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('does pass prop open to Snackbar', () => {
    const open = wrapper.find(Snackbar).props().open;
    expect(open).toEqual(true);
  });

  it('has correct value of message inside Alert', () => {
    const label = wrapper.find(Alert).text();
    expect(label).toEqual('Hello world');
  });
});
