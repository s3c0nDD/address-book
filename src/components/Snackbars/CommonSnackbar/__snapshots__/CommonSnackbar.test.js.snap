// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonSnackbar shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(Snackbar))
  open={true}
>
  <WithStyles(ForwardRef(Alert))
    severity="info"
    variant="filled"
  >
    Hello world
  </WithStyles(ForwardRef(Alert))>
</WithStyles(ForwardRef(Snackbar))>
`;
